pkg/cmd/cli/describe/describer.go

```diff
@@ -111,30 +111,25 @@ func (d *BuildDescriber) Describe(namespace, name string) (string, error) {
 		}
 		// Create the time object with second-level precision so we don't get
 		// output like "duration: 1.2724395728934s"
-		t := util.Now().Rfc3339Copy()
-		if build.StartTimestamp != nil && build.CompletionTimestamp != nil {
-			// time a build ran from pod creation to build finish or cancel
-			formatString(out, "Duration", build.CompletionTimestamp.Sub(build.StartTimestamp.Rfc3339Copy().Time))
-		} else if build.CompletionTimestamp != nil && build.Status == buildapi.BuildStatusCancelled {
-			// time a build waited for its pod before ultimately being canceled before that pod was created
-			formatString(out, "Duration", fmt.Sprintf("waited for %s", build.CompletionTimestamp.Sub(build.CreationTimestamp.Rfc3339Copy().Time)))
-		} else if build.CompletionTimestamp != nil && build.Status != buildapi.BuildStatusCancelled {
-			// for some reason we never saw the pod enter the running state, so we don't know when it
-			// "started", so instead print out the time from creation to completion.
-			formatString(out, "Duration", build.CompletionTimestamp.Sub(build.CreationTimestamp.Rfc3339Copy().Time))
-		} else if build.StartTimestamp == nil && build.Status != buildapi.BuildStatusCancelled {
-			// time a new build has been waiting for its pod to be created so it can run
-			formatString(out, "Duration", fmt.Sprintf("waiting for %s", t.Sub(build.CreationTimestamp.Rfc3339Copy().Time)))
-		} else if build.CompletionTimestamp == nil {
-			// time a still running build has been running in a pod
-			formatString(out, "Duration", fmt.Sprintf("running for %s", t.Sub(build.StartTimestamp.Rfc3339Copy().Time)))
-		}
+		formatString(out, "Duration", describeBuildDuration(build))
 		formatString(out, "Build Pod", buildutil.GetBuildPodName(build))
 		describeBuildParameters(build.Parameters, out)
 		return nil
 	})
 }
 
+func describeBuildDuration(build *buildapi.Build) string {
+	t := util.Now().Rfc3339Copy()
+	if build.StartTimestamp == nil {
+		// time a new build has been waiting for its pod to be created so it can run
+		return fmt.Sprintf("waiting for %v", t.Sub(build.CreationTimestamp.Rfc3339Copy().Time))
```

Comment:
you're going to output "waiting for XXXs" for builds that were canceled before starting.  Should output "waited"